src/index.ts

```diff
@@ -22,117 +20,113 @@ export type Context = object
+  credits_reserved: bigint // u64
+  credits_accepted: bigint // u64
+  timestamp: bigint // u64, Set this to 0n - the actual value will be set by TigerBeetle server
 }
 
-export type CreateAccount = Account & {
-  timestamp: bigint // u64
+export enum AccountFlags {
+  linked = (1 << 0),
+  debits_must_not_exceed_credits = (1 << 1),
+  credits_must_not_exceed_debits = (1 << 2)
 }
 
 export enum CreateAccountError {
-    exists = 1,
-    exists_with_different_unit,
-    exists_with_different_limits,
-    exists_with_different_custom_field,
-    exists_with_different_flags,
-    reserved_field_custom,
-    reserved_field_padding,
-    reserved_field_timestamp,
-    reserved_flag_padding,
-    exceeds_debit_reserved_limit,
-    exceeds_debit_accepted_limit,
-    exceeds_credit_reserved_limit,
-    exceeds_credit_accepted_limit,
-    debit_reserved_limit_exceeds_debit_accepted_limit,
-    credit_reserved_limit_exceeds_credit_accepted_limit,
+  linked_event_failed = 1,
+  exists,
+  exists_with_different_user_data,
+  exists_with_different_reserved_field,
+  exists_with_different_unit,
+  exists_with_different_code,
+  exists_with_different_flags,
+  exceeds_credits,
+  exceeds_debits,
+  reserved_field,
+  reserved_flag_padding,
 }
 
-export type CreateAccountResult = {
+export type CreateAccountsError = {
   index: number,
-  error: CreateAccountError,
+  code: CreateAccountError,
 }
 
-export type CreateTransfer = {
+export type Transfer = {
   id: bigint, // u128
   debit_account_id: bigint, // u128
   credit_account_id: bigint, // u128
-  custom_1: bigint, // u128
-  custom_2: bigint, // u128
-  custom_3: bigint, // u128
-  flags: bigint, // u64
-  amount: bigint, // u64
+  user_data: bigint, // u128
+  reserved: Buffer, // [32]u8
   timeout: bigint, // u64, in nano-seconds
+  code: number, // u32 accounting system code to describe the type of transfer (e.g. settlement)
+  flags: number, // u32
+  amount: bigint, // u64,
+  timestamp: bigint, // u64, Set this to 0n - the actual value will be set by TigerBeetle server
 }
 
-export enum CreateTransferFlags {
-  accept = (1 << 0),
-  reject = (1 << 1),
-  auto_commit = (1 << 2)
+export enum TransferFlags {
+  linked = (1 << 0),
+  two_phase_commit = (1 << 1),
+  condition = (1 << 2) // whether or not a condition will be supplied
 }
 
 export enum CreateTransferError {
-  exists = 1,
+  linked_event_failed = 1,
+  exists,
   exists_with_different_debit_account_id,
   exists_with_different_credit_account_id,
-  exists_with_different_custom_fields,
+  exists_with_different_user_data,
+  exists_with_different_reserved_field,
+  exists_with_different_code,
   exists_with_different_amount,
   exists_with_different_timeout,
   exists_with_different_flags,
   exists_and_already_committed_and_accepted,
   exists_and_already_committed_and_rejected,
-  reserved_field_custom,
-  reserved_field_timestamp,
+  reserved_field,
   reserved_flag_padding,
-  reserved_flag_accept,
-  reserved_flag_reject,
```

Comment:
These still exist in tigerbeetle:
https://github.com/coilhq/tigerbeetle/blob/main/src/tigerbeetle.zig#L233-L234